automation/taskcluster/lib/tasks.py

```diff
@@ -0,0 +1,155 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import datetime
+import json
+import taskcluster
+
+
+class TaskBuilder(object):
+    def __init__(self, task_id, repo_url, branch, commit, owner, source):
+        self.task_id = task_id
+        self.repo_url = repo_url
+        self.branch = branch
+        self.commit = commit
+        self.owner = owner
+        self.source = source
+
+    def build_task(self, name, description, command, dependencies=[],
+                   artifacts={}, scopes=[], routes=[], features={},
+                   worker_type='github-worker'):
+        created = datetime.datetime.now()
+        expires = taskcluster.fromNow('1 year')
```

Comment:
This function is not called on PRs, right? If it is, I'd suggest to downtime to `1 month` which is what `try` defines.